### PR TITLE
Change the API priority

### PR DESCRIPTION
**Pull request by CaoDonghui123**
#### What type of PR is this?

/kind api-change
#### What this PR does / why we need it:
TODO (mortent): priority should change after 1.21. See https://github.com/kubernetes/kubernetes/pull/95718#discussion_r520969477
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by pacoxu** [comment]
/cc @mortent

You may follow https://github.com/kubernetes/kubernetes/commit/8574e3e3f4d14125014d22d9311ac1765863fcb6 to fix the test failures.

test/integration/etcd/data.go need updated as well.

**Comment by pacoxu** [comment]
To stop CI from running tests, you can convert PR to Draft. And `/test pull-kubernetes-unit ` to run a specified test to check the current commit.

**Comment by fedebongio** [comment]
assign @kubernetes/api-reviewers 
/remove-sig api-machinery

File: pkg/registry/policy/podsecuritypolicy/storage/storage_test.go [hunk 1]
```diff
@@ -72,40 +71,6 @@ func validNewPodSecurityPolicy() *policy.PodSecurityPolicy {
 	}
 }
 
-func TestCreate(t *testing.T) {
```
Comment (liggitt) [inline pkg/registry/policy/podsecuritypolicy/storage/storage_test.go:75]: restore these?

**Review by liggitt — COMMENTED** [review]
~~lgtm other than restoring the dropped tests~~

edit: this breaks the podsecuritypolicy API. see https://github.com/kubernetes/kubernetes/pull/102943#discussion_r656784722

File: pkg/registry/policy/podsecuritypolicy/storage/storage_test.go (hunk 1 above)
Comment (CaoDonghui123) [reply pkg/registry/policy/podsecuritypolicy/storage/storage_test.go:75]: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/102943/pull-kubernetes-unit/1405733896372359168
I tried to fix these failed by removing TestCreate() and TestUpdate()

Comment (liggitt) [reply pkg/registry/policy/podsecuritypolicy/storage/storage_test.go:75]: oh... the failing tests mean that this change completely breaks the podsecuritypolicy API. That resource doesn't have a v1 type, so switching the default for policy to v1 means that that API fails to encode.

You need to do something like the following to tell the server to keep storing PodSecurityPolicy objects in v1beta1:

```diff
diff --git a/pkg/kubeapiserver/default_storage_factory_builder.go b/pkg/kubeapiserver/default_storage_factory_builder.go
index 31479d77646..4802f4cf07d 100644
--- a/pkg/kubeapiserver/default_storage_factory_builder.go
+++ b/pkg/kubeapiserver/default_storage_factory_builder.go
@@ -62,6 +62,7 @@ func NewStorageFactoryConfig() *StorageFactoryConfig {
 
 	resources := []schema.GroupVersionResource{
 		apisstorage.Resource("csistoragecapacities").WithVersion("v1beta1"),
+		policy.Resource("podsecuritypolicies").WithVersion("v1beta1"),
 	}
 
 	return &StorageFactoryConfig{
```

Comment (CaoDonghui123) [reply pkg/registry/policy/podsecuritypolicy/storage/storage_test.go:75]: thank you for your help
I will fix it

Comment (CaoDonghui123) [reply pkg/registry/policy/podsecuritypolicy/storage/storage_test.go]: look like the test failed still there

**Comment by ritazh** [comment]
@CaoDonghui123 can you pls rebase?

File: pkg/controlplane/storageversionhashdata/data.go [hunk 2]
```diff
@@ -77,9 +77,9 @@ var GVRToStorageVersionHash = map[string]string{
 	"networking.k8s.io/v1/ingressclasses":                                  "l/iqIbDgFyQ=",
 	"node.k8s.io/v1/runtimeclasses":                                        "WQTu1GL3T2Q=",
 	"node.k8s.io/v1beta1/runtimeclasses":                                   "WQTu1GL3T2Q=",
-	"policy/v1/poddisruptionbudgets":                                       "6BGBu0kpHtk=",
-	"policy/v1beta1/poddisruptionbudgets":                                  "6BGBu0kpHtk=",
-	"policy/v1beta1/podsecuritypolicies":                                   "khBLobUXkqA=",
+	"policy/v1/poddisruptionbudgets":                                       "EVWiDmWqyJw=",
+	"policy/v1beta1/poddisruptionbudgets":                                  "EVWiDmWqyJw=",
+	"policy/v1beta1/podsecuritypolicies":                                   "bAz9Pn7x8wg=",
```
Comment (liggitt) [inline pkg/controlplane/storageversionhashdata/data.go:82]: podsecuritypolicies must not change, there's no v1 version to encode to (I would not expect this test to pass)

**Pull request closed without merging**